source/tests/consistent/fitting/test_ener.py

```diff
@@ -0,0 +1,180 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import unittest
+from typing import (
+    Any,
+    Tuple,
+)
+
+import numpy as np
+
+from deepmd.dpmodel.fitting.invar_fitting import EnergyFittingNet as EnerFittingDP
+from deepmd.env import (
+    GLOBAL_NP_FLOAT_PRECISION,
+)
+
+from ..common import (
+    INSTALLED_PT,
+    INSTALLED_TF,
+    CommonTest,
+    parameterized,
+)
+from .common import (
+    FittingTest,
+)
+
+if INSTALLED_PT:
+    import torch
+
+    from deepmd.pt.model.task.ener import EnergyFittingNet as EnerFittingPT
+    from deepmd.pt.utils.env import DEVICE as PT_DEVICE
+else:
+    EnerFittingPT = object
+if INSTALLED_TF:
+    from deepmd.tf.fit.ener import EnerFitting as EnerFittingTF
+else:
+    EnerFittingTF = object
+from deepmd.utils.argcheck import (
+    fitting_ener,
+)
+
+
+@parameterized(
+    (True, False),  # resnet_dt
+    ("float64", "float32"),  # precision
+    (True, False),  # distinguish_types
+    (0, 1),  # numb_fparam
+)
+class TestEner(CommonTest, FittingTest, unittest.TestCase):
+    @property
+    def data(self) -> dict:
+        (
+            resnet_dt,
+            precision,
+            distinguish_types,
+            numb_fparam,
+        ) = self.param
+        return {
+            "neuron": [5, 5, 5],
+            "resnet_dt": resnet_dt,
+            "precision": precision,
+            "numb_fparam": numb_fparam,
+            "seed": 20240217,
+        }
+
+    @property
+    def skip_tf(self) -> bool:
+        (
+            resnet_dt,
+            precision,
+            distinguish_types,
+            numb_fparam,
+        ) = self.param
+        # TODO: distinguish_types
+        return not distinguish_types or CommonTest.skip_pt
+
+    @property
+    def skip_pt(self) -> bool:
+        (
+            resnet_dt,
+            precision,
+            distinguish_types,
+            numb_fparam,
+        ) = self.param
+        # TODO: float32 has bug
```

Comment:
@anyangml It should be fixed

Reply:
what's the issue here?

Reply:
it seems that float 32 does not pass the consistency test. what is the difference? do we reduce the tolerance accordingly when using float32?

Reply:
> do we reduce the tolerance accordingly when using float32?

It just complains that a float64 tensor cannot be operated with a float32 tensor.

Reply:
please raise an issue for it. Thanks!

Reply:
I found only pt has the bug. The dp backend had the problem of tolerance, and I fixed that.